### PR TITLE
Terminal: Close terminal buffer with EXIT signal

### PR DESCRIPTION
**Pull request by HollowMan6**
When the server is about to close, it will send "Closing" to the front-end. Then on receiving that infomation, the front-end will close using `window.close()`

### CODE REVIEW

File: app/terminal/index.html [hunk 1]
```diff
@@ -46,6 +46,10 @@
          });
          const searchAddon = new SearchAddon.SearchAddon();
 
+         function get_status() {
+             return is_closed;
```
Comment (MatthewZMD) [inline app/terminal/index.html:50]: is_closed undefined

File: app/terminal/buffer.py [hunk 2]
```diff
@@ -88,6 +94,12 @@ def destroy_buffer(self):
             # NOTE: We need delete QWebEnginePage manual, otherwise QtWebEngineProcess won't quit.
             self.buffer_widget.web_page.deleteLater()
             self.buffer_widget.deleteLater()
+            
+    def on_close(self):
+        if self.buffer_widget.execute_js("get_status();"):
+            self.timer=None
+            self.destroy_buffer()
+            self.eval_in_emacs.emit('''(eaf-request-kill-buffer "'''+self.buffer_id+'''")''')
```
Comment (MatthewZMD) [inline app/terminal/buffer.py:102]: self.close_buffer()应该就够了

File: app/terminal/index.html [hunk 3]
```diff
@@ -120,7 +124,14 @@
...
 
              window.addEventListener("resize", sendSizeToServer);
-         }
+             
+             function change_status() {
+                    is_closed = true
+                    console.log("CLOSED")
+             }
+             
+             socket.addEventListener("close",change_status);
```
Comment (MatthewZMD) [inline app/terminal/index.html:133]: 整个可以简化成window.close()

**Comment by MatthewZMD** [comment]
这整个写复杂了，关键点是window.close()，还有server.js文件

**Comment by HollowMan6** [comment]
感谢提醒，一开始没想到用window.close()可以直接关闭

**Comment by MatthewZMD** [comment]
可以的

**Pull request merged**